modules/ipp/ipp.go

```diff
@@ -0,0 +1,71 @@
+package ipp
+
+import (
+	"bytes"
+	"encoding/binary"
+	//"io"
+	"net"
+)
+
+type Connection struct {
+	Conn net.Conn
+}
+
+//func ReadResponse(body *io.ReadCloser) *ScanResults {
+//	result := &ScanResults{}
+//
+//}
+
+func reverse(b []byte) []byte {
+	for i := 0; i < len(b)/2; i++ {
+		b[i], b[len(b)-1-i] = b[len(b)-1-i], b[i]
+	}
+	return b
+}
+
+//FIXME: Clean this up, and use binary package to handle endianness in the correct order
+//FIXME: Make sure uint isn't messing anything up here, since they should be signed.
+func attributeByteString(syntaxTag byte, name string, value string) []byte {
+	//special bytestring denoting value syntax
+	b := []byte{syntaxTag}
+	l := make([]byte, 2)
+	binary.PutUvarint(l, uint64(len(name)))
```

Comment:
Do they actually use golang's varint format (high bit -> more bytes remaining), or is this just assuming len(name) < 127`? In the latter case, it is probably more appropriate to just put a fixed 8-bit size.

 - [ ] Add link to RFC section describing the encoding